ghost/core/core/server/data/db/backup.js

```diff
@@ -45,6 +45,12 @@ const readBackup = async (filename) => {
  * @returns {Promise<String>}
  */
 const backup = async function backup(options = {}) {
+    // do not create backup if disabled in config (this is intended for large customers who will OOM node)
+    if (config.get('disableJSBackups')) {
+        logging.info('Database backup is disabled in Ghost config');
+        return '';
```

Comment:
Do we know for sure that consumers of this module can handle an empty string return value?

Reply:
Afaict the consumers don't even use the string, but it's clearly structured to return a file path in case that needs to be served to the front end. I made the return a null string because it shouldn't reject - that would likely interrupt the upstream code - and generally the return type is `Promise<String>`. 

These are the callers and the use cases I can make out:
- [Deleting a user](https://github.com/TryGhost/Ghost/blob/3c56005d44b0cbb3e6a9e72ff3cf38c40569f9c7/ghost/core/core/server/services/Users.js#L151); filename not used
- [Migration before hook](https://github.com/TryGhost/Ghost/blob/3c56005d44b0cbb3e6a9e72ff3cf38c40569f9c7/ghost/core/core/server/data/migrations/hooks/migrate/before.js#L6); filename not used but **I'm also not sure how this doesn't break migrations unless it doesn't run for large customers, because it should always fail for them?**
- [Delete all content/backup content routes](https://github.com/TryGhost/Ghost/blob/3c56005d44b0cbb3e6a9e72ff3cf38c40569f9c7/ghost/core/core/server/web/api/endpoints/admin/routes.js#L220-L224); these are only manually accessed with Ghost API requests

I don't see any issue with these use cases. I am curious about the migrations one though, which maybe you could shed some light on?

Reply:
@9larsons Yeah, that line is disabled on Pro 🙂

Reply:
Ah right!